nats.go

```diff
@@ -2834,9 +2840,10 @@ func (nc *Conn) subscribe(subj, queue string, cb MsgHandler, ch chan *Msg) (*Sub
 
 // NumSubscriptions returns active number of subscriptions.
 func (nc *Conn) NumSubscriptions() int {
-	nc.mu.Lock()
-	defer nc.mu.Unlock()
-	return len(nc.subs)
+	nc.mu.RLock()
```

Comment:
Prefer the defer version, and all of these are not critical perf wise, so let's revert these.

Reply:
done.